db-client-java/src/main/java/com/eventstore/dbclient/GrpcClient.java

```diff
@@ -182,13 +183,26 @@ private boolean discover(UUID previousId, Optional<Endpoint> candidate) {
     private boolean loadServerFeatures() {
         try {
             serverInfo = ServerFeatures.getSupportedFeatures(settings, channel);
+            serverInfoLoaded = true;
             return true;
         } catch (ServerFeatures.RetryableException e) {
             logger.warn("An exception happened when fetching server supported features. Retrying connection attempt.", e);
             return false;
         }
     }
 
+    public ServerVersion getServerVersion() throws RuntimeException {
+        if (!serverInfoLoaded) {
+            throw new RuntimeException("server info not loaded");
```

Comment:
Considering that logic was not present before, why are we throwing an exception in this case?

Reply:
this is for the case where the client hasn't connected to any node yet, so no server version is available/known yet. i'm not sure if that's the best way to handle it

Reply:
That shouldn't happen. If it does it now, it means some regression was introduced with recent PRs.

Reply:
say somebody does the following:
```
SingleNodeClient client = new SingleNodeClient();
client.getServerVersion();
```

given that messageLoop() is executed asynchronously in the constructor, isn't there a time window where it's possible that the server info isn't yet loaded?

Reply:
I would suggest doing it as we do with other clients (Go or the Rust clients for example). Meaning getting the server version should be an async operation. I truly believed that was the case since the beginning, but maybe an overlooked some changes made in recent PRs.

If it takes too much time for you to fix it, keep it the way you did and I'll change it in another PR. Please create an issue, so I remember.